functions/gene_go_annotation.scm

```diff
@@ -1,11 +1,9 @@
 (define* (gene-go-annotation gene_nodes namespace #:optional (parents 0))
     (let (
-      [hs (make-hash-table)]
     )
-  (for-each (lambda (gene) 
-      (hash-set! hs gene (find-go-term gene namespace parents))
+  (ListLink (flatten (map (lambda (gene) 
```

Comment:
@Habush I found empty result testing it. I think the reason is result from (find-go-term) for each gene are not collected.  What I recommend will be:

(define* (gene-go-annotation gene_nodes namespace #:optional (parents 0))
  (let ([result '()])
  (for-each (lambda (gene)
  (set! result (append result (find-go-term gene namespace parents)))
  )gene_nodes)

 (ListLink result)
)) 

As the output of  (find-go-term) is already a ListLink, you only need to collect the result for each gene in a list and return the result as a ListLink. Hence we will have nested ListLinks in the result.

Reply:
@Habush going through the results while bench marking, I am getting Go terms under namespace 'molecular_function' while I request for (list "cellular_component"  "biological_process")
Can you please check?

Reply:
@tanksha Can you send me an example that reproduces an empty result? In my test, it returned a `ListLink` of the results.

Reply:
It actually works, I don't know why it happened yesterday. But there is a serious issue that given any namespace, all three are being retrieved. You can even ask for empty string namespace and get all GO terms in three of the namespaces. I think it is the OrLink which is creating this problem, can you please check?

Reply:
use the sample dataset [here](https://gist.github.com/tanksha/a2b8d6f6f771e5b3a0dc9fb8a97a3c9e) and try the following 
- (gene-go-annotation (list "SPAG9") (list "cellular_component"))
check the result, it also contains GO terms under molecular_function and bological_process

- (gene-go-annotation (list "SPAG9") (list ""))
You can even pass empty string as a namespace and the result contains all GO terms under three of the namespaces.